pilot/pkg/model/endpointshards_test.go

```diff
@@ -38,37 +38,68 @@ func TestUpdateServiceAccount(t *testing.T) {
 		{
 			name:      "added new endpoint",
 			shardKey:  c1Key,
-			endpoints: append(cluster1Endppoints, &IstioEndpoint{Address: "10.172.0.3", ServiceAccount: "sa1"}),
+			endpoints: append(cluster1Endppoints, &IstioEndpoint{Addresses: []string{"10.172.0.3"}, ServiceAccount: "sa1"}),
+			expect:    false,
+		},
+
+		{
+			name:      "added new endpoint with multiple addresses",
+			shardKey:  c1Key,
+			endpoints: append(cluster1Endppoints, &IstioEndpoint{Addresses: []string{"10.171.0.1", "2001:1::1"}, ServiceAccount: "sa1"}),
 			expect:    false,
 		},
 		{
 			name:      "added new sa",
 			shardKey:  c1Key,
-			endpoints: append(cluster1Endppoints, &IstioEndpoint{Address: "10.172.0.3", ServiceAccount: "sa2"}),
+			endpoints: append(cluster1Endppoints, &IstioEndpoint{Addresses: []string{"10.172.0.3"}, ServiceAccount: "sa2"}),
+			expect:    true,
+		},
+		{
+			name:      "added new sa of an endpoint with multiple addresses",
+			shardKey:  c1Key,
+			endpoints: append(cluster1Endppoints, &IstioEndpoint{Addresses: []string{"10.172.0.3", "2001:1::3"}, ServiceAccount: "sa2"}),
 			expect:    true,
 		},
 		{
 			name:     "updated endpoints address",
 			shardKey: c1Key,
 			endpoints: []*IstioEndpoint{
-				{Address: "10.172.0.5", ServiceAccount: "sa1"},
-				{Address: "10.172.0.2", ServiceAccount: "sa-vm1"},
+				{Addresses: []string{"10.172.0.5"}, ServiceAccount: "sa1"},
+				{Addresses: []string{"10.172.0.2"}, ServiceAccount: "sa-vm1"},
+			},
+			expect: false,
+		},
+		{
+			name:     "updated endpoints multiple addresses",
```

Comment:
I do not see tests being updated to use the dual-stack feature flag. Will this be honored even without that?

Reply:
Hi @kfaseela, I suggest that you can check the tests adding in below code files for multiple addresses

1. pilot/pkg/serviceregistry/serviceentry/controller_test.go
2. pilot/pkg/serviceregistry/serviceentry/conversion_test.go
3. pilot/pkg/serviceregistry/serviceentry/store_test.go